botorch_community/models/vbll_helper.py

```diff
@@ -5,531 +5,475 @@
 # LICENSE file in the root directory of this source tree.
 
 """
-The following code is from the repository vbll (https://github.com/VectorInstitute/vbll)
-which is under the MIT license.
+Variational Bayesian Last Layers — enhanced port.
+
+Original: vbll (https://github.com/VectorInstitute/vbll), MIT license.
 Paper: "Variational Bayesian Last Layers" by Harrison et al., ICLR 2024
+
+Enhancements:
+- Use torch consistently for numerics (no numpy for log/dtypes).
+- Device/dtype aware operations (torch.as_tensor with matching device/dtype).
+- Improved numerical stability with torch.clamp.
+- Convenience helpers: predictive sampling, posterior mean/covariance.
+- Clearer typing and docstrings.
+- Minor robustness fixes and shape checks.
 """
 
 from __future__ import annotations
 
 from dataclasses import dataclass
-from typing import Callable
+from typing import Callable, Union, Optional
 
-import numpy as np
 import torch
 import torch.nn as nn
-
-from botorch.logging import logger
 from torch import Tensor
+from botorch.logging import logger
 
 
-def tp(M):
+def tp(M: Tensor) -> Tensor:
+    """Transpose the last two dimensions of a tensor."""
     return M.transpose(-1, -2)
 
 
 class Normal(torch.distributions.Normal):
-    def __init__(self, loc: Tensor, chol: Tensor):
-        """Normal distribution.
+    """
+    Diagonal Gaussian wrapper. 'scale' is interpreted as the std-dev vector.
+    """
 
-        Args:
-            loc (_type_): _description_
-            chol (_type_): _description_
-        """
-        super().__init__(loc, chol)
+    def __init__(self, loc: Tensor, scale: Tensor):
+        # ensure shape broadcastability but keep behavior identical to torch.Normal
+        super().__init__(loc, scale)
```

Comment:
I'm confused by this comment. It looks to me like there's no need for an init method at all since it's just calling the super class's init.